### PR TITLE
Specifiable Netty EventLoopGroup and executor

### PR DESCRIPTION
**Pull request by aquadrini**
I've added a few changes mainly regarding Netty behaviour:

- Added support to specify an existing EventLoopGroup to reuse.
- Added support for Epoll ELG and Channel.
- Allow to execute callbacks in same ELG threads, specifying a null executor in builder, for efficient execution of very short lived tasks.
- Dependencies upgrade.
- Reduce memory footprint via AtomicReferenceFieldUpdater usage

### CODE REVIEW

File: folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java [hunk 1]
```diff
@@ -69,27 +73,26 @@
...
   private final long timeoutMillis;
   private final int maxSetLength;
 
-  private final AtomicReference<String> disconnectReason = new AtomicReference<>(null);
+  private static final
+    AtomicReferenceFieldUpdater<DefaultRawMemcacheClient, String> DISCONNECT_REASON_UPDATER =
+          AtomicReferenceFieldUpdater.newUpdater(DefaultRawMemcacheClient.class,
+                  String.class, "disconnectReason");
+
+  private volatile String disconnectReason;
```
Comment (spkrka) [inline folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java]: Is this change really worth the added code complexity? The main purpose of `AtomicReferenceFieldUpdater` seems to be to reduce memory usage, but the number of active `DefaultRawMemcacheClient` is typically fairly small.

File: folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java [hunk 2]
```diff
@@ -114,7 +119,7 @@
       decoder = new AsciiMemcacheDecoder(charset);
     }
 
-    final ChannelHandler initializer = new ChannelInitializer<Channel>() {
+    final ChannelHandler initializer = new ChannelInitializer() {
```
[the remark this answers is not in the document]
Comment (aquadrini) [reply folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java]: It is, sorry. It's just an IDE issue that slipped through. I've fixed it.

File: folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java (hunk 1 above)
Comment (aquadrini) [reply folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java]: Hi! Thanks for your review! I've just added it to follow Netty style of using references, to reduce memory footprint as much as possible in every case. It's true that client instance count will be usually low, so it's ok for me if you prefer to rollback the change.

Comment (spkrka) [reply folsom/src/main/java/com/spotify/folsom/client/DefaultRawMemcacheClient.java]: Yes, I think we would prefer that. If you revert that, we will be happy to squash-merge and make a new release

**Comment by aquadrini** [comment]
Great! I've just reverted the AtomicReferenceFieldUpdater changes. Thanks!

**Comment by spkrka** [comment]
Thanks! I have released 1.1.0 now

**Pull request merged**